### PR TITLE
git-flow CLI

### PR DESCRIPTION
**Pull request by MadLittleMods**
Fixes https://github.com/smith-kyle/nodegit-flow/issues/9

git-flow CLI.

Still WIP but would love some discussion/suggestions on the direction this should go. If we want to suss this out through chat, I am [available on Gitter](https://gitter.im/MadLittleMods) but would be great to create a `nodegit-flow` specific room to keep the discussion out in the open. This could end up as a separate package if we don't want the CLI added here.

The reason for this over some of the other CLI options out there is so that there is an option on Windows that doesn't require cygwin. Plus JavaScript based :+1:

``` sh
git-flow init

git-flow feature start foo
git-flow feature finish foo

git-flow release start 0.1.0
git-flow release finish 0.1.0

git-flow hotfix start 0.1.1
git-flow hotfix finish 0.1.1
```

> 
> 
> http://danielkummer.github.io/git-flow-cheatsheet/
## Todo
- [ ] Better logging
  - [ ] Should we use `console.log`?
- [ ] Figure out why `finish` `mergeCommit.id()` doesn't work as documented 😕 
- [ ] `publish`
- [ ] `push` (maybe)

### CODE REVIEW

**Comment by smith-kyle** [comment]
This is great! Personally, I don't mind adding the cli to this project (as long as it never causes the API to change) but I can think of a couple reasons why you might want to make it a separate thing. 

With your own repo/package, you'd be able to give the project a more concise description and name so potential users would have an easier time finding your project. Secondly, most users of nodegit-flow are going to be using it either through the cli or JS, not both. That means sifting through docs will be a bit tougher since you'd have to first find the section that refers to the API you're using.

Overall I think it'd be less confusing to users if we kept the two separate. But like I said I'm open to the idea. Let me know if there are any advantages I'm not seeing.

File: src/bin/git-flow.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env devtool
```
Comment (MadLittleMods) [inline src/bin/git-flow.js]: *update to `#!/usr/bin/env node`*

File: src/bin/git-flow.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      const { name } = argv;
+      return getRepo()
+        .then((repo) => {
+          if(type === 'feature') {
+            return Git.Flow.startFeature(repo, name);
+          }
+          else if(type === 'release') {
+            return Git.Flow.startRelease(repo, name);
+          }
+          else if(type === 'hofix') {
```
Comment (MadLittleMods) [inline src/bin/git-flow.js]: typo `hotfix`

**Comment by Thatkookooguy** [comment]
@MadLittleMods don't know if this helps, but I started working on something similar myself: https://github.com/Kibibit/kibibit-cli

It's currently not stable (I'm in the middle of writing it :-)), and it's also coupled with github (by a single file called `kb-github`, which you can implement whatever cloud version control you want if you replace that file)

It's not vanilla nodegit-flow (I do some extra stuff, like creating a repo on init if you're not already inside a git repo), but might suite your needs

**Comment by MadLittleMods** [comment]
Closing as it looks like this was built into `git` (Git for Windows) since 2015-12-8 https://github.com/git-for-windows/git/releases/tag/v2.6.4

> As of Git for Windows 2.6.4, GitFlow (AVH edition) is included, so you're all done.
>
> https://github.com/petervanderdoes/gitflow-avh/wiki/Installing-on-Windows#git-for-windows

Works for me :+1:

```
$ git --version
git version 2.7.1.windows.1
$ git flow -h
usage: git flow <subcommand>

Available subcommands are:
   init      Initialize a new git repo with support for the branching model.
   feature   Manage your feature branches.
   bugfix    Manage your bugfix branches.
   release   Manage your release branches.
   hotfix    Manage your hotfix branches.
   support   Manage your support branches.
   version   Shows version information.
   config    Manage your git-flow configuration.
   log       Show log deviating from base branch.

Try 'git flow <subcommand> help' for details.
```

**Pull request closed without merging**